dialer.go

```diff
@@ -475,14 +475,17 @@ func lookupHost(ctx context.Context, address string, resolver Resolver) (string,
 		// if the resolver doesn't return anything, we'll fall back on the provided
 		// address instead
 		if len(resolved) > 0 {
-			resolvedHost, resolvedPort := splitHostPort(resolved[0])
+			resolvedHost, resolvedPort, _ := net.SplitHostPort(resolved[0])
 
 			// we'll always prefer the resolved host
-			host = resolvedHost
+			if len(resolvedHost) == 0 && len(resolvedPort) == 0 {
```

Comment:
Just to keep it consistent with the `resolvedPort != ""` check below:
```suggestion
			if lresolvedHost != "" && resolvedPort != "" {
```